### ISSUE TITLE
ControlStatementRule failing on acceptable use of parentheses

### ISSUE THREAD

**Issue by mattrubin** [realm/SwiftLint#189]
`ControlStatementRule` is showing a warning on the following piece of code. I think this is an acceptable use of parens in an `if` statement, and the rule should be updated to accept it.

```swift
if (minValue...maxValue).contains(value) {
    // Do something...
}
```
> Control Statement Parentheses Violation: if statements shouldn't wrap their conditionals in parentheses. (control_statement)

**Comment by jpsim** [comment]
Agreed. PRs welcome if you'd like to address this "false positive".

**Comment by mmorier** [comment]
I also notice this issue for the following case where the cause is the same.
```swift
if (isA || isB) && (isC || isD) {
    // Do something...
}
```
I'll try to find a way to fix it and make a PR.

**Comment by jpsim** [comment]
> I'll try to find a way to fix it and make a PR.

Great! Thanks for your help.
